modules/util/util.go

```diff
@@ -63,3 +64,52 @@ func Min(a, b int) int {
 func IsEmptyString(s string) bool {
 	return len(strings.TrimSpace(s)) == 0
 }
+
+// NormalizeEOL will convert Windows (CRLF) and Mac (CR) EOLs to UNIX (LF)
+func NormalizeEOL(input []byte) []byte {
+	var right, left, pos int
+	if right = bytes.IndexByte(input, '\r'); right == -1 {
+		return input
+	}
+	length := len(input)
+	tmp := make([]byte, length)
+	for left < length && input[left] == '\n' {
+		left++
+	}
+	if left != 0 {
+		copy(tmp[pos:pos+left], input[0:left])
+		pos = left
+		right -= left
+	}
+
+	if left < length {
+		if input[left] == '\n' {
+			left++
+			right--
+		}
```

Comment:
We can drop this check. We know that input[left] is not a `\n` from line 76.

Reply:
In fact now that we have unfurled the initial step we can drop lines 76-84 and lines 86-89

Reply:
And you can drop the `if left < length {` test because the only way that could happen would cause `right == -1`.

Reply:
See https://github.com/go-gitea/gitea/pull/8925#issuecomment-552804972

Reply:
👍 @zeripath done.